torch_geometric/nn/conv/message_passing.py

```diff
@@ -227,6 +242,8 @@ def propagate(self, edge_index: Adj, size: Size = None, **kwargs):
             **kwargs: Any additional data which is needed to construct and
                 aggregate messages, and to update node embeddings.
         """
+        if self.__explain__:
+            self.decomposed_layers = 1  # assuming decomposed layers is passed to init
```

Comment:
```suggestion
        decomposed_layers = 1 if self.__explain__ else self.decomposed_layers 
```

Reply:
I agree with you very much and I have modified it.